python/ModelTools.py

```diff
@@ -53,6 +53,7 @@ def doSet(self,name,vars):
         if self.options.bin: self.out.defineSet(name,vars)
         else: self.out.write("%s = set(%s);\n" % (name,vars));
     def doObj(self,name,type,X):
+        if self.out.obj(name): return 1 # Do not try to reproduce constraint term if it exists
```

Comment:
I'd rather not change this, since at the moment it can trap other possible mistakes of object redefinition.